lib/Dialect/LWE/IR/LWEDialect.cpp

```diff
@@ -0,0 +1,54 @@
+#include "include/Dialect/LWE/IR/LWEDialect.h"
+
+#include "include/Dialect/LWE/IR/LWEAttributes.h"
+#include "llvm/include/llvm/ADT/TypeSwitch.h"            // from @llvm-project
+#include "mlir/include/mlir/IR/DialectImplementation.h"  // from @llvm-project
+
+// Generated definitions
+#include "include/Dialect/LWE/IR/LWEDialect.cpp.inc"
+#define GET_ATTRDEF_CLASSES
+#include "include/Dialect/LWE/IR/LWEAttributes.cpp.inc"
+
+namespace mlir {
+namespace heir {
+namespace lwe {
+
+void LWEDialect::initialize() {
+  addAttributes<
+#define GET_ATTRDEF_LIST
+#include "include/Dialect/LWE/IR/LWEAttributes.cpp.inc"
+      >();
+}
+
+LogicalResult BitFieldEncodingAttr::verifyEncoding(
+    ArrayRef<int64_t> shape, Type elementType,
+    ::llvm::function_ref<::mlir::InFlightDiagnostic()> emitError) const {
+  if (!elementType.isa<IntegerType>()) {
```

Comment:
nit: should they also be signless? I don't think I'd understand why a signed integer would be encoded like this.

Reply:
Done